SingularityService/src/main/java/com/hubspot/singularity/resources/RequestResource.java

```diff
@@ -171,34 +190,18 @@ public SingularityRequestParent bounce(@ApiParam("The request ID to bounce") @Pa
       checkBadRequest(currentActiveSlaveCount >= requiredSlaveCount, "Not enough active slaves to successfully complete a bounce of request %s (minimum required: %s, current: %s). Consider deploying, or changing the slave placement strategy instead.", requestId, requiredSlaveCount, currentActiveSlaveCount);
     }
 
-    final Optional<Boolean> skipHealthchecks = bounceRequest.isPresent() ? bounceRequest.get().getSkipHealthchecks() : Optional.<Boolean> absent();
-
-    Optional<String> message = Optional.absent();
-    Optional<String> actionId = Optional.absent();
-
-    if (bounceRequest.isPresent()) {
-      actionId = bounceRequest.get().getActionId();
-      message = bounceRequest.get().getMessage();
-
-      if (bounceRequest.get().getDurationMillis().isPresent() && !actionId.isPresent()) {
```

Comment:
Ah, now that I look again - are we now saying it's impossible to have unlimited bounce?

If so, what will you do if someone enters a negative value for this configuration? Need to have validation on it or allow people to set it to 0/negative to indicate that they don't want to use this feature.

In which case, we'd only set actionId if there was a duration set.